### DISCUSSION TITLE
Need help with parsing graphQL queries

### DISCUSSION THREAD

**Discussion by kapishmalik** [graphql-java/graphql-java#3213] [General]
Hi,

I need to parse graphQL queries for which I need a parser for the same. If I need that I need to add graphql-java as dependency which increases my jar size as it gets all the dependencies. Is there any way I can get only parser part of it or any way to exclude guava/antlr dependencies in shadow JAR.

**Comment by dondonz** [comment]
Our parser uses ANTLR so there wouldn't be a way to separate the parser from the ANTLR dependency.

I'm not sure what your size restrictions are. We've tried to keep the dependencies to an absolute minimum. We only shade antlr-runtime which is a subset of the full ANTLR library and only limited parts of Guava

**Reply by dondonz** [reply]
To add more background on why we shaded antlr-runtime initially, we had some reports from users that they had clashing ANTLR versions, because ANTLR is one of those libraries that turns up everywhere. This prevented some people from using older versions of graphql-java as they were unable to resolve the dependency.

On balance we thought it would be better to shade antlr-runtime to avoid the version clash problem. Although it does increase the shadow jar size, we tried to keep the overhead as small as possible by only shading the smaller runtime dependency.

**Comment by kapishmalik** [comment]
@dondonz is there any other alternative I can use to parse graphql queries?

**Reply by kapishmalik** [reply]
We just want to parse graphql queries and compare them. Apart from that, we don't need anything.

**Comment by bbakerman** [comment]
>graphql-java as dependency which increases my jar size as it gets all the dependencies. 

What environment are you running in that you truly care about how big the jar is?  In 2023 that is often not a valid concern

Remember also that at runtime, you only load the classes you use.  The guava we shade is just the immutable collections (whcih we use in the parse code) and antler plus our code.
